src/Data/Vec/Base.agda

```diff
@@ -262,25 +262,21 @@ allFin _ = tabulate id
 splitAt : ∀ m {n} (xs : Vec A (m + n)) →
           ∃₂ λ (ys : Vec A m) (zs : Vec A n) → xs ≡ ys ++ zs
 splitAt zero    xs                = ([] , xs , refl)
-splitAt (suc m) (x ∷ xs)          with splitAt m xs
-splitAt (suc m) (x ∷ .(ys ++ zs)) | (ys , zs , refl) =
-  ((x ∷ ys) , zs , refl)
+splitAt (suc m) (x ∷ xs) =
+  let (ys , zs , eq) = splitAt m xs in ((x ∷ ys) , zs , cong (x ∷_) eq)
 
 take : ∀ m {n} → Vec A (m + n) → Vec A m
-take m xs          with splitAt m xs
-take m .(ys ++ zs) | (ys , zs , refl) = ys
+take m xs = proj₁ (splitAt m xs)
 
 drop : ∀ m {n} → Vec A (m + n) → Vec A n
-drop m xs          with splitAt m xs
-drop m .(ys ++ zs) | (ys , zs , refl) = zs
+drop m xs = proj₁ (proj₂ (splitAt m xs))
 
 group : ∀ n k (xs : Vec A (n * k)) →
         ∃ λ (xss : Vec (Vec A k) n) → xs ≡ concat xss
 group zero    k []                  = ([] , refl)
-group (suc n) k xs                  with splitAt k xs
-group (suc n) k .(ys ++ zs)         | (ys , zs , refl) with group n k zs
-group (suc n) k .(ys ++ concat zss) | (ys , ._ , refl) | (zss , refl) =
-  ((ys ∷ zss) , refl)
+group (suc n) k xs with splitAt k xs
+group (suc n) k xs | (ys , zs , eq) with group n k zs
+group (suc n) k xs | (ys , ._ , eq) | (zss , eq′) = ((ys ∷ zss) , trans eq (cong (ys ++_) eq′))
```

Comment:
Again, I'd consider using one of 
* nested, irrefutable, `with` here
```agda
group (suc n) k xs
  with ys , zs , eq-split ← splitAt k xs
  with zss , eq-group ← group n k zs
  =  (ys ∷ zss) , trans eq-split (cong (ys ++_) eq-group)
```
* else a pattern-matching `let`
```agda
group (suc n) k xs  = 
  let ys , zs , eq-split = splitAt k xs in
  let zss , eq-group     = group n k zs in
   (ys ∷ zss) , trans eq-split (cong (ys ++_) eq-group)
```

In each case avoiding redundant parentheses around tuples...

Reply:
I like that `let` version - I'll use that one.